internal/app/api/repositories/user_repo.go

```diff
@@ -99,6 +99,18 @@ func (r *MongoRepository) Login(ctx context.Context, UserLogin types.UserLogin)
 
 }
 
+func (r *MongoRepository) FindByID(ctx context.Context, id string) (*types.User, error) {
```

Comment:
Just create a method to get the field need to verify, no need to fetch all the fields